### PR TITLE
ADD: Resizing Images and Accepting alternative inputs

### PR DESCRIPTION
**Pull request by sihrc**
Added Ruby Image Resizing
Added Base64 String and Filepath Input

### CODE REVIEW

File: lib/indico/image.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    # Base 64 handling
+    if regex == nil
+      raise Exception.new("String is not a valid absolute filepath or base64 string")
+    else
+      return ChunkyPNG::Image.from_data_url("data:image/png;base64," + str.gsub("data:image/png;base64," ,""))
+    end
+  end
+
+  def self.handle_array_input(image)
+    # Handles properly formatting and loading array of pixels
```
Comment (madisonmay) [inline lib/indico/image.rb]: Should we do the same thing as python here and let people know that array inputs won't be supported long-term?

File: lib/indico/image.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    # Returns Integer encoding of RGB value used by ChunkyPNG
+    return [
+      ChunkyPNG::Color.r(value),
+      ChunkyPNG::Color.g(value),
+      ChunkyPNG::Color.b(value)
+    ]
+  end
+
+  def self.load_image(filepath)
+    # Loads and populates properly formatted array from filepath
```
Comment (madisonmay) [inline lib/indico/image.rb]: We should be using the base64 format over the array format whenever possible.

**Comment by madisonmay** [comment]
Nice work finding `oily_png`, works like a charm.

**Comment by sihrc** [comment]
@madisonmay  I'm also annoyed by oily_png's warnings.

**Comment by madisonmay** [comment]
We should silence our warnings during the tests.

File: lib/indico/image.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        im_array.push(preprocess(_image, size, false))
+      end
+
+      return im_array
+    end
+
+    if image.class == String
+      decoded_image = handle_string_input(image)
+    elsif image.class == Array
+      warn "Warning! Array input as image will be deprecated in the next major release.\n Consider using filepaths or base64 encoded strings"
```
Comment (sihrc) [inline lib/indico/image.rb]: @madisonmay These warnings?

File: lib/indico/image.rb (hunk 1 above)
Comment (madisonmay) [inline lib/indico/image.rb]: Awesome.

**Comment by madisonmay** [comment]
Looks good to me!  Merging to development.

**Pull request merged**